internal/config/config.go

```diff
@@ -408,32 +335,32 @@ func parsePeer(p peer) (*Peer, error) {
 		RouterID:      routerID,
 		NodeSelectors: nodeSels,
 		Password:      password,
-		BFDProfile:    p.BFDProfile,
-		EBGPMultiHop:  p.EBGPMultiHop,
+		BFDProfile:    p.Spec.BFDProfile,
+		EBGPMultiHop:  p.Spec.EBGPMultiHop,
 	}, nil
 }
 
-func parseAddressPool(p addressPool, bgpCommunities map[string]uint32) (*Pool, error) {
+func addressPoolFromCR(p metallbv1beta1.AddressPool, bgpCommunities map[string]uint32) (*Pool, error) {
 	if p.Name == "" {
 		return nil, errors.New("missing pool name")
 	}
 
 	ret := &Pool{
-		Protocol:      p.Protocol,
-		AvoidBuggyIPs: p.AvoidBuggyIPs,
+		Protocol:      Proto(p.Spec.Protocol), // TODO CRDs: validate proto
```

Comment:
If the addresspool is coming from a k8s custom resource,
that has the Kubebuilder validation for enum: layer2; bgp.
why another validation is required here?

Reply:
that was a reminder for me because I don't like unchecked castings. I can remove the TODO, this version of CRDs / translation is transient and this part will go away in any case.